api/fetch_tags.py

```
@@ -0,0 +1,84 @@
+        'python', 'javascript', 'typescript', 'react', 'node', 
+        'html', 'css', 'go', 'rust', 'docker', 'java', 'c++', 
+        'backend', 'frontend'
+    ]
+    
+    # 2. Setup the search query parameters for GitHub API
+    query = f'repo:{repo_name} label:"good first issue" state:open'
+    encoded_query = urllib.parse.quote(query)
+    url = f'https://api.github.com/search/issues?q={encoded_query}&per_page=30'
+    
+    # 3. Construct the HTTP request payload with a custom User-Agent header
+    req = urllib.request.Request(url, headers={'User-Agent': 'gsoc-org-finder'})
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)